### DISCUSSION TITLE
What is "BPR" alignment?

### DISCUSSION THREAD

**Discussion by kamfaima** [Azure/ResourceModules#3184] [General]
Hello!

I see there's been a change in the past 3 days labelled "BPR alignment". This led me to wonder is this in preparation to populate the (sparse) bicep public registry (https://azure.github.io/bicep-registry-modules/ and https://github.com/Azure/bicep-registry-modules) with the modules available in this repo?

If so, awesome(!) and when would this be happening and if not, why not, as the public registry is pretty slim pickings as far as resources go!

Note: I can see this has been discussed before (https://github.com/Azure/ResourceModules/discussions/837) but that was in December 2021. Was hoping to see more sharing/reusing of resources between the two repos.

Thanks!

**Comment by eriqua** [comment]
Hello @kamfaima, correct. 
We are currently working on aligning the CARML module structure to the one required by the Bicep Public Registry, with the end goal of publishing our modules there.
We cannot share a due date for this work to be completed, but you can refer to issue #3131 for tracking the progress.
Also note the disclaimer on the [home page](https://github.com/Azure/ResourceModules#upcoming-breaking-changes).

**Reply by AlexanderSehr** [reply]
As @eriqua stated, it is ongoing. However - the major breaking changes we have to implement (i.e. file & folder renamings) will hopefully be done very soon. Then there are a couple other changes to implement such as new files (e.g., `metadata.json`) and finally the probalby most time consuming one: The User-Defined-Types. But those have to be released first, so ... we're working on what we can ;)

**Reply by kamfaima** [reply]
That's amazing, thanks for the response. I'm looking forward to the day where we'll have the option of using the BPR modules in our code!
